src/main/scala/com/typesafe/scalalogging/LoggerTakingImplicit.scala

```diff
@@ -0,0 +1,83 @@
+package com.typesafe.scalalogging
+
+import org.slf4j.{ Marker, Logger => Underlying }
+
+trait CanLog[A] { this: Serializable =>
+  def logMessage(originalMsg: String, a: A): String
+  def afterLog(a: A): Unit = ()
+}
+
+@SerialVersionUID(957385465L)
```

Comment:
why a SerialVersionUID and why Serializable?

Reply:
let's remove this

Reply:
Just `SerialVersionUID` or also `Serializable`?

Responding to an earlier question, it just mirrors whats in "regular" `Logger` + to allow Spark users to use it

Reply:
Ah ok makes sense